src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -1398,6 +1420,75 @@ public Type getReturnType(String invocationId) {
         }
     }
 
+    // We don't have reconnect yet, but this helps align the Java client with the .NET client
+    // and hopefully make it easier to implement reconnect in the future
+    private final class ReconnectingConnectionState {
+        private final Logger logger;
+        private final Lock lock = new ReentrantLock();
+        private ConnectionState state;
+        private HubConnectionState hubConnectionState = HubConnectionState.DISCONNECTED;
+        public Completable startTask;
+        public TransportEnum currentTransport;
```

Comment:
Can both startTask and currentTransport be moved to ConnectionState?

Reply:
startTask should stay since its lifetime is greater than the ConnectionStates lifetime, can probably move the transport enum

Reply:
Are you sure the startTask's lifetime is greater than ConnectionState's lifetime? It looks to me that the ConnectionState is set immediate in HubConnection.start() inside the same lock that transitions the HubConnection to the CONNECTING state and is unset inside the same lock that transitions the HubConnection back to the DISCONNECTED at which point the startTask would no longer be used.